internal/provider/provider.go

```diff
@@ -69,9 +77,15 @@ func (p *ConductoroneProvider) Configure(ctx context.Context, req provider.Confi
 	ServerURL := data.ServerURL.ValueString()
 	ClientID := data.ClientID.ValueString()
 	ClientSecret := data.ClientSecret.ValueString()
+	TenantDomain := data.TenantDomain.ValueString()
 
 	if ServerURL == "" {
-		ServerURL = "https://{tenantDomain}.conductor.one"
+		if TenantDomain == "" {
+			url := strings.Split(ClientID, "@")[1]
+			ServerURL = fmt.Sprintf("https://%s", url)
+		} else {
+			ServerURL = fmt.Sprintf("https://%s.conductor.one", TenantDomain)
+		}
 	}
 
 	opt, err := sdk.WithTenantCustom(ServerURL)
```

Comment:
like this logic is backwards, isn't it?

Shouldn't we only set this option, IF TenantDomain or ServerURL is set.  And then let the SDK fallback to parsing the client ID which it already doesn't internally?